presto-tests/src/main/java/io/prestosql/tests/sql/TestTable.java

```diff
@@ -13,21 +13,48 @@
  */
 package io.prestosql.tests.sql;
 
+import java.util.Optional;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import static java.lang.String.format;
+
 public class TestTable
         implements AutoCloseable
 {
+    public static final String TABLE_NAME_PLACEHOLDER = "{TABLE_NAME}";
+
     private final SqlExecutor sqlExecutor;
     private final String name;
 
     private static final AtomicInteger instanceCounter = new AtomicInteger();
 
     public TestTable(SqlExecutor sqlExecutor, String namePrefix, String createDdlTemplate)
+    {
+        this(sqlExecutor, namePrefix, createDdlTemplate, Optional.empty());
+    }
+
+    public TestTable(SqlExecutor sqlExecutor, String namePrefix, String createDdlTemplate, Optional<String> data)
     {
         this.sqlExecutor = sqlExecutor;
         this.name = namePrefix + "_" + instanceCounter.incrementAndGet();
-        sqlExecutor.execute(createDdlTemplate.replace("{TABLE_NAME}", this.name));
+        sqlExecutor.execute(createDdlTemplate.replace(TABLE_NAME_PLACEHOLDER, this.name));
+        if (!data.isPresent()) {
+            return;
+        }
+        try {
+            sqlExecutor.execute(format("INSERT INTO %s VALUES %s", name, data.get()));
+        }
+        catch (Exception e) {
+            try {
+                close();
+            }
+            catch (Exception innerException) {
+                if (e != innerException) {
```

Comment:
Could it really be the same exception instance?

Reply:
Yes, I know. I think I can call it a `Presto convention`.